src/NuGet.Services.AzureSearch/Db2AzureSearch/NewPackageRegistrationProducer.cs

```diff
@@ -34,6 +34,7 @@ public class NewPackageRegistrationProducer : INewPackageRegistrationProducer
 
         public async Task ProduceWorkAsync(
             ConcurrentBag<NewPackageRegistration> allWork,
+            HashSet<string> excludedPackages,
             CancellationToken cancellationToken)
```

Comment:
Instead of accepting the `excludedPackages` in this method, consider calling `IAuxiliaryFileClient.LoadExcludedPackagesAsync` in this method to get the excluded packages.